keps/sig-multicluster/1645-multi-cluster-services-api/specification.md

```diff
@@ -0,0 +1,212 @@
+
+### 2.1 - Definitions
+
+This proposal is intended as an extension of the [cluster-local Kubernetes DNS specification](https://github.com/kubernetes/dns/blob/master/docs/specification.md), and inherits its definitions from section 2.1  with the addition of the following:
+
+hostname = as already defined in the [cluster-local Kubernetes DNS specification](https://github.com/kubernetes/dns/blob/master/docs/specification.md), this refers (in brief): in order of precedence, to a) the value of the endpoint's `hostname` field, or b) a unique, system-assigned identifier for the endpoint. Of importance to highlight is that since the [default hostname of an endpoint is the Pod's `metadata.name` field](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields), this will likely often be the podname, but not always, and implementations must prefer a directly specified `hostname` value.
+
+clusterset = as defined in [KEP-1645: Multi-Cluster Services API](README.md): “A placeholder name for a group of clusters with a high degree of mutual trust and shared ownership that share services amongst themselves. Membership in a clusterset is symmetric and transitive. The set of member clusters are mutually aware, and agree about their collective association. Within a clusterset, [namespace sameness](https://github.com/kubernetes/community/blob/master/sig-multicluster/namespace-sameness-position-statement.md) applies and all namespaces with a given name are considered to be the same namespace.”
+
+`<clusterset-zone>` = domain for multi-cluster services in the clusterset, which must be `clusterset.local`; as this may become configurable in the future, this specification refers to it by the placeholder `<clusterset-zone>`, but per the MCS API it currently must be defined to be `clusterset.local`. 
+
+ClusterSetIP / `<clusterset-ip>` / clusterset IP = as defined in [KEP-1645: Multi-Cluster Services API](README.md): “A non-headless ServiceImport is expected to have an associated IP address, the clusterset IP, which may be accessed from within an importing cluster. This IP may be a single IP used clusterset-wide or assigned on a per-cluster basis, but is expected to be consistent for the life of a ServiceImport from the perspective of the importing cluster. Requests to this IP from within a cluster will route to backends for the aggregated Service.”
+
+Cluster ID / `<clusterid>` = the cluster id stored in the `id.k8s.io ClusterProperty` as described in [KEP-2149: ClusterId for ClusterSet identification](../2149-clusterid/README.md). Though this can be any valid DNS label, the recommended value is  a kube-system namespace uid ( such as `721ab723-13bc-11e5-aec2-42010af0021e`). For ease of KEP readability, this document uses human readable names `clusterA` and `clusterB` to represent the cluster IDs of two clusters in a ClusterSet.
```

Comment:
nit: s/clusterA/cluster-a/  ?

Reply:
Fixed in [c4e0549](https://github.com/kubernetes/enhancements/pull/2577/commits/c4e05493b6917d0fe1d8d451367259061eac704a)